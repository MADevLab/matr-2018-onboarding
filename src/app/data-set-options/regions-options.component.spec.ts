import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { MatSelectModule, MatSelect } from '@angular/material/select';
import { MatOption } from '@angular/material';

import { ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';

import { RegionsOptionsComponent } from './regions-options.component';

describe('The RegionsOptionsComponent', () => {
  let component: RegionsOptionsComponent;
  let fixture: ComponentFixture<RegionsOptionsComponent>;

  beforeEach(async () => {
    TestBed.configureTestingModule({
      imports: [
        CommonModule,
        ReactiveFormsModule,
        NoopAnimationsModule,
        MatSelectModule
      ],
      declarations: [ RegionsOptionsComponent ]
    });

    await TestBed.compileComponents();
    fixture = TestBed.createComponent(RegionsOptionsComponent);
    component = fixture.componentInstance;
  });

  it('is defined', () => {
    expect(component).toBeDefined();
  });

  it('emits changes to the regions', () => {
    const spy = jest.spyOn(component.changeRegions, 'emit');

    component.regions = _regions;
    component.onRegionsChange(['NE', 'SO']);

    expect(spy).toHaveBeenCalled();
    expect(spy).toMatchSnapshot();
  });
});

const _regions = [
  {
    id: 'NE',
    title: 'Northeast',
    regions: ['CT', 'DC', 'DE']
  },
  {
    id: 'MW',
    title: 'Midwest',
    regions: ['WI']
  },
  {
    id: 'SO',
    title: 'South',
    regions: ['AL', 'AR']
  },
  {
    id: 'WE',
    title: 'West',
    regions: ['WA', 'WY']
  }
];
