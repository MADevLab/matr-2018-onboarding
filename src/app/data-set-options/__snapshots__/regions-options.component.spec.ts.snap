// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The RegionsOptionsComponent emits changes to the regions 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        Object {
          "id": "NE",
          "regions": Array [
            "CT",
            "DC",
            "DE",
          ],
          "title": "Northeast",
        },
        Object {
          "id": "SO",
          "regions": Array [
            "AL",
            "AR",
          ],
          "title": "South",
        },
      ],
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;
